### PR TITLE
Add model types

### PR DESCRIPTION
**Pull request by BrandonRomano**

### CODE REVIEW

**Comment by BrandonRomano** [comment]
cc @tmilewski @antonelli

File: README.md [hunk 1]
```diff
@@ -59,8 +60,93 @@ Interface models are the models as they will be represented in the API request.
...
+[DELETE] /model/:id/nested-model/:nested-model-id     // Deletes a single model (Delete)
+```
+
+Where `model` is the name of model(b) in plural form, and `nested-model` is the name of model(a) in plural form.  Model(a) and model(b) can be switched around in this case, so try to pick the one that makes the most sense for your application.
+
+For example, if we had an interface model for `people` and `dogs` (and we lived in a world where people can have multiple dogs, and dogs can have multiple owners) our urls could either all start with `/people/{person-id}/dogs` or `/dogs/{dog-id}/people`, depending on which made the most sense in your application.
+
 ## Spec Out RESTful Methods
 
+> These methods are really have already been already decided by work done in the [previous section](#determine-interface-model-types), but here we get to decide which "subsets" of endpoints to add for each model.  We also get to specify further information (like parameters) with help us further understand the API.
```
Comment (BrandonRomano) [inline README.md]: Whoop, this is not a coherent sentence... I need to fix this

File: README.md [hunk 2]
```diff
@@ -59,8 +60,93 @@ Interface models are the models as they will be represented in the API request.
...
+#### Many-to-Many Models
+
+Many to many models are models who are responsible for associating two other models (model(a) to model(b)). These models can contain additional information about the association, but that is optional.
+
+> These are synonymous to many-to-many models in a relational database system.  Most always if your database model is a many-to-many model, your interface model should also be many-to-many.  Any deviation from this should be discussed with the team.
+
+The endpoints of a Many-to-Many model will be a subset of the following:
+
+```
+[POST]   /model/:id/nested-model/:nested-model-id     // Creates a single model (Create)
```
Comment (tmilewski) [inline README.md]: Should it be `record` over `model`? (Across all endpoints)

Comment (BrandonRomano) [inline README.md]: Yeah, I prefer that verbiage :100: -- will make a change!

**Comment by BrandonRomano** [comment]
Merging.  Will update later if there are any suggestions!

**Pull request merged**